Filter allocs properly by taskgroup

djenriquez opened the pull request:
Prior to this change, CPU/Memory calculations were for an entire job were being compared to taskgroup policies. This caused taskgroups to incorrectly scale.

PTAL

Commit:
Filter allocs properly by taskgroup

kainoaseto commented on client/nomad.go:
```diff
@@ -256,11 +256,18 @@ func (c *nomadClient) EvaluateJobScaling(jobName string, jobScalingPolicies []*s
+				allocs = append(allocs, alloc)
+			}
+		}
+
```
Probably should put this filtering into the `c.GetJobAllocations` function since it looks like that's where allocation filtering occurs

kainoaseto replied:
Line 304 would let this check just be a one liner

Commit:
Use better func name, add more deployment statuses to deployment check

Commit:
Filter deeper for taskgroup name

Commit:
Some refactoring

kainoaseto commented on client/nomad.go:
```diff
@@ -105,15 +105,15 @@ func (c *nomadClient) MostUtilizedResource(alloc *structs.ClusterCapacity) {
 	case gsp.Tasks.Resources.MemoryPercent:
-		gsp.ScalingMetric = ScalingMetricMemory
+		return ScalingMetricMemory
 	}
```
What happens if neither of the cases are met? Looks like we need a return statement here

Commit:
Cleanup logic

Commit:
Add go module files and updated noamd version to 0.9.6, updated version to 1.2.0 and resolved build to only build linux/amd64 since that's the only binary that gets copied into the final image

kainoaseto committed:
Add go module files and updated noamd version to 0.9.6, updated version to 1.2.0 and resolved build to only build linux/amd64 since that's the only binary that gets copied into the final imagedd go module files and updated noamd version to 0.9.6, updated version to 1.2.0 and resolved build to only build lin
ux/amd64 since that's the only binary that gets copied into the final image

~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
~
"~/Workspace/code/go/src/github.com/djenriquez/replicator/.git/COMMIT_EDITMSG" 15L, 592C
:wq


kainoaseto committed:
Merge branch 'fix/taskgroup-scaling' of github.com:djenriquez/replicator into fix/taskgroup-scaling

kainoaseto reviewed (commented):
LGTM
